app/models/container_deployment.rb

```diff
@@ -6,7 +6,368 @@ class ContainerDeployment < ApplicationRecord
+  end
+
+  def public_ip?(ip)
+    require "socket"
+    return ip unless Addrinfo.tcp(ip, 80).ipv4_private?
+  end
+
+  def generate_automation_params(params)
+    parameters = { # for all
+      :deployment_type   => kind.to_s,
+      :deployment_method => method_type,
+      :deployment_id     => id,
+      :provider_name     => params["provider_name"],
+      :containerized     => containerized,
+      :rhsub_sku         => rhsm_auth.rhsm_sku
+    }
+    if method_type.include? "provision"
+      node_template = VmOrTemplate.find(params["nodes_creation_template_id"])
+      master_template = VmOrTemplate.find(params["masters_creation_template_id"])
+      parameters_provision = {
+        :ssh_public_key        => ssh_auth.public_key,
+        :provision_provider_id => deployed_on_ems.id,
+        :masters_provision     => {
+          :number_of_vms  => container_nodes_by_role("master").count,
+          :templateFields => {
+            "guid" => master_template.guid, "name" => master_template.name
+          },
+          :vmFields       => vm_fields("masters", params),
+          :requester      => {
+            'owner_email'      => 'something@example.com',
+            'owner_first_name' => 'temp',
+            'owner_last_name'  => 'temp'
+          },
+          :tags           => nil,
+          :ws_values      => {
+            "ssh_public_key" => ssh_auth.public_key
+          }
+        },
+        :nodes_provision       => {
+          :number_of_vms  => num_of_nodes,
+          :templateFields => {
+            "guid" => node_template.guid, "name" => node_template.name
+          },
+          :vmFields       => vm_fields("nodes", params),
+          :requester      => {
+            'owner_email'      => 'something@example.com',
+            'owner_first_name' => 'temp',
+            'owner_last_name'  => 'temp'
+          },
+          :tags           => nil,
+          :ws_values      => {
+            "ssh_public_key" => ssh_auth.public_key
+          }
+        },
+      }
+      parameters.merge parameters_provision
+    else
+      parameters_existing = { # non-managed + managed-existing
+        :deployment_master => roles_addresses("deployment_master"),
+        :masters           => roles_addresses("master"),
+        :nodes             => roles_addresses("node")
+      }
+      parameters.merge parameters_existing
+    end
+  end
+
+  def vm_fields(type, params)
+    options = {}
+    options["customization_template_id"] = CustomizationTemplate.find_by_name("SSH key addition template").id
+    options["vm_name"] = type.include?("master") ? params["master_base_name"] : params["node_base_name"]
+    options["vm_memory"] = params[type + "_vm_memory"] if params[type + "_vm_memory"]
+    options["cpu"] = params[type + "_cpu"] if params[type + "_cpu"]
+    options["instance_type"] = params[type + "_instance_type"] if params[type + "_instance_type"]
+    options
+  end
+
+  def generate_ssh_keys
```

Comment:
@blomquisg Yeah, that's really a wrapper around Net::SSH than a utility class.  It feels like this key generator should be a simple library in lib.  If this lived in MiqSshUtilV2, it would be out of place other than the SSH in the name.